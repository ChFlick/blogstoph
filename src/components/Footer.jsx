import React, { Fragment } from 'react';

const Footer = (props) => (
    <Fragment>
        <hr className="bar--light-grey" />
        <div className="copyright">Copyright Â© Christoph Flick 2018</div>
    </Fragment>
);

export default Footer;
