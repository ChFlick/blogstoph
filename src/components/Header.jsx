import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import { startLogout } from '../actions/auth';

const DEFAULT_HEADER_CLASSES = 'header header--fixed-top';

export class Header extends React.Component {
    state = {
        headerClasses: DEFAULT_HEADER_CLASSES
    }

    componentDidMount = () => {
        window.addEventListener('scroll', this.handleScroll);
    };

    componentWillUnmount = () => {
        window.removeEventListener('scroll', this.handleScroll);
    };

    handleScroll = () => {
        const scrollPosPrev = this.state.scrollPos;
        const scrollPos = window.scrollY;
        let headerClasses = 'header header--fixed-top';

        if (scrollPos > 80) {
            headerClasses = 'header header--fixed';
            if (scrollPos < scrollPosPrev) {
                headerClasses += ' header--visible';
            }

            this.setState(() => ({ headerClasses }));
        } else if (scrollPos === 0) {
            this.setState(() => ({ headerClasses: DEFAULT_HEADER_CLASSES }));
        }

        this.setState(() => ({ scrollPos }));
    };

    createAreaSwitchButton = () =>
        this.props.private ?
            <Link className="button button--link button--header" to="/dashboard">Visitor Area</Link> :
            <Link className="button button--link button--header" to="/editor/dashboard">Editor Area</Link>;


    render() {
        return (
            <header id="header" className={this.state.headerClasses} onScroll={this.handleScroll}>
                <div className="content-container">
                    <div className="header__content">
                        <Link className="header__title" to="/dashboard">
                            <h1>Blogstoph</h1>
                        </Link>
                        {this.props.isAuthenticated ? (
                            <div>
                                {this.createAreaSwitchButton()}
                                <button className="button button--link button--header" onClick={this.props.startLogout}>Logout</button>
                            </div>
                        ) : (
                                <Link className="button button--link" to="/login">Login</Link>
                            )}
                    </div>
                </div>
            </header>
        );
    };
};

const mapStateToProps = (state) => ({
    isAuthenticated: !!state.auth.uid
});

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
