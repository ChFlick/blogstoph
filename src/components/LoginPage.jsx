import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Blogstoph Login</h1>
            <p>Log in to edit your posts.</p>
            <button className="button" onClick={startLogin}>Login with Google</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin('google'))
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
