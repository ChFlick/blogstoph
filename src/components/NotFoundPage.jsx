import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
    <Fragment>
        <header className="page-header">
            <h1>404 Not Found</h1>
        </header>
        <section className="content-container align-center">
            <p>The page you were looking for could not be found.</p>
            <Link to="/">Click here to go back to the home page</Link>
        </section>
    </Fragment>
);

export default NotFoundPage;
