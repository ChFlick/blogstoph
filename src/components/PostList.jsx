import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import moment from 'moment';

const isNotLastElement = (index, numberOfPosts) => {
    return index !== numberOfPosts - 1;
};

export const PostList = (props) => (
    <div className="content-container">
        {
            props.posts.length === 0 ? (
                <p>There are no posts available</p>
            ) : (
                    props.posts.map((post, index) =>
                        <Fragment key={post.id}>
                            <div
                                className="post-preview"
                            >
                                <h3 className="post-preview__title">{post.title}</h3>
                                <p className="post-preview__subtitle">{post.content}</p>
                                <span className="post-preview__data">
                                    Posted by {post.author} at {moment(post.date).format('DD. MMM YYYY')}
                                </span>
                            </div>
                            {isNotLastElement(index, props.posts.length) && <hr className="post-preview__separator" />}
                        </Fragment>
                    )
                )
        }
    </div>
);

const mapStateToProps = (state) => ({
    posts: (state.posts || []).filter((post = {}) => post.published === true)
});

export default connect(mapStateToProps)(PostList);
