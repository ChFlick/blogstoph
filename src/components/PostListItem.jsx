import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

const PostListItem = (props) => (
    <Fragment>
        <Link
            className="post-preview"
            to={`post/${props.post.id}`}
        >
            <h3 className="post-preview__title">{props.post.title}</h3>
            <p className="post-preview__subtitle">{props.post.content}</p>
        </Link>
        <span className="post-preview__data">
            Posted by {props.post.author} at {moment(props.post.date).format('DD. MMM YYYY')}
        </span>
    </Fragment>
);

export default PostListItem;
