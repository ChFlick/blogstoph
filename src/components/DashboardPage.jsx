import React from 'react';

import PostList from './PostList';

const DashboardPage = () => (
    <div>
        <h1>Dashboard</h1>
        <PostList />
    </div>
);

export default DashboardPage;
