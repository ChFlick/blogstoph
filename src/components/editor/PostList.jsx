import React, { Fragment } from 'react';
import { connect } from 'react-redux';

import PostListItem from './PostListItem';

export const PostList = (props) => (
    <Fragment>
        {
            props.posts.length === 0 ? (
                <p>There are no posts available</p>
            ) : (
                props.posts.map((post) => (
                    <PostListItem post={post} key={post.id} />
                ))
            )
        }
    </Fragment>
);

const mapStateToProps = (state) => ({
    posts: state.posts || []
});

export default connect(mapStateToProps)(PostList);
