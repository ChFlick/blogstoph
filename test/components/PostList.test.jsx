import React from 'react';
import { shallow } from 'enzyme';

import { PostList } from '../../src/components/PostList';

describe('post list', () => {
    test('should render correctly', () => {
        const wrapper = shallow(<PostList />);

        expect(wrapper).toMatchSnapshot();
    });
});
