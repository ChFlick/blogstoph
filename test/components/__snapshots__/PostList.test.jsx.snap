// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post list should render correctly with posts 1`] = `
<div
  className="content-container"
>
  <React.Fragment
    key="1"
  >
    <div
      className="post-preview"
    >
      <h3
        className="post-preview__title"
      >
        Some Title
      </h3>
      <p
        className="post-preview__subtitle"
      >
        This is a Content of a Post...
      </p>
      <span
        className="post-preview__data"
      >
        Posted by 
        Author of 1
         at 
        01. Jan 1970
      </span>
    </div>
    <hr
      className="post-preview__separator"
    />
  </React.Fragment>
  <React.Fragment
    key="2"
  >
    <div
      className="post-preview"
    >
      <h3
        className="post-preview__title"
      >
        Some Title 2
      </h3>
      <p
        className="post-preview__subtitle"
      >
        This is a second Content of a Post...
      </p>
      <span
        className="post-preview__data"
      >
        Posted by 
        Author of 2
         at 
        01. Jan 1970
      </span>
    </div>
    <hr
      className="post-preview__separator"
    />
  </React.Fragment>
  <React.Fragment
    key="3"
  >
    <div
      className="post-preview"
    >
      <h3
        className="post-preview__title"
      >
        Some Title 3
      </h3>
      <p
        className="post-preview__subtitle"
      >
        This is a third Content of a Post...
      </p>
      <span
        className="post-preview__data"
      >
        Posted by 
        Author of 3
         at 
        01. Jan 1970
      </span>
    </div>
  </React.Fragment>
</div>
`;

exports[`post list should render correctly without posts 1`] = `
<div
  className="content-container"
>
  <p>
    There are no posts available
  </p>
</div>
`;
